x-pack/plugin/core/src/main/resources/ecs-dynamic-mappings.json

```diff
@@ -0,0 +1,198 @@
+{
+  "template": {
+    "mappings": {
+      "date_detection": false,
```

Comment:
As this is included after `logs@custom`, it should only include dynamic templates as other mappings can't be overridden in `logs@custom`. For example, setting `"date_detection": true` in `logs@custom` wouldn't have any effect.

Reply:
I was able to trace a conversation about why date_detection is false : https://github.com/elastic/elasticsearch/pull/57629/files/099c782ced2c9619ffcc3a48f88b39c9ce1a8efc#r435033616

Reply:
> As this is included after logs@custom, it should only include dynamic templates as other mappings can't be overridden in logs@custom. For example, setting `"date_detection": true` in `logs@custom` wouldn't have any effect.

No, this is the point, my assumption is that as long as we include a default `string -> date` mapping, it implicitly disables the date detection as any string input that doesn't match and dynamic template pattern and doesn't have an explicit mapping, will be mapped to `keyword`. I believe that setting `"date_detection": true` will have no effect either way, but I can verify.

> I was able to trace a conversation about why date_detection is false 

This is a very general discussion about it that basically assumes that the default enablement of this feature is not the right. 
But assuming we want to put it elsewhere- why data-stream mappings and not logs mappings? The linked conversation is about logging mappings

Reply:
> I believe that setting "date_detection": true will have no effect either way, but I can verify.

I think you're right but if someone wants to override the string -> keyword dynamic template, they can do so in logs@custom. But they can't set date_detection to true as the setting in ecs-dynamic-mappings would take precedence.

We could add `date_detection: false` in `logs-mappings`.

Reply:
> I think you're right but if someone wants to override the string -> keyword dynamic template, they can do so in logs@custom. But they can't set date_detection to true as the setting in ecs-dynamic-mappings would take precedence.

I still think we won't provide a good way out of this. IINM, the only way to override the `string -> keyword` mapping is with another `string -> XXX` mapping, which leaves `data_detection` implicitly disabled.

Anyway,  I think that moving it to `logs-mappings` makes sense as a statement that we prefer a different default for this feature in logs data streams, regardless of the fact that it is implicitly disabled anyway.
I will move it.